src/event_logger.cpp

```diff
@@ -60,17 +60,90 @@ void EventLogger::createAsyncPelWithInventoryCallout(
     const std::optional<std::string> i_symFru,
     const std::optional<std::string> i_procedure)
 {
-    // TODO, implementation needs to be added.
-    (void)i_errorType;
-    (void)i_severity;
-    (void)i_callouts;
-    (void)i_fileName;
-    (void)i_funcName;
-    (void)i_internalRc;
-    (void)i_userData1;
-    (void)i_userData2;
     (void)i_symFru;
     (void)i_procedure;
+    try
+    {
+        if (i_callouts.empty())
+        {
+            logging::logMessage(
+                "Callout information is missing to create PEL.");
+            return;
+        }
+
+        if (m_errorMsgMap.find(i_errorType) == m_errorMsgMap.end())
+        {
+            logging::logMessage(
+                "PEL error type not found in the error message map.");
+            return;
+        }
+
+        const std::string& l_message = m_errorMsgMap.at(i_errorType);
+
+        const std::string& l_severity =
+            (m_severityMap.find(i_severity) != m_severityMap.end()
+                 ? m_severityMap.at(i_severity)
+                 : m_severityMap.at(types::SeverityType::Error));
+
+        std::string l_description("Description field missing in user data");
+
+        if (i_userData1 && ((*i_userData1).first.compare("DESCRIPTION") ==
+                            constants::STR_CMP_SUCCESS))
+        {
+            l_description = (*i_userData1).second;
+        }
+
+        sd_bus* l_sdBus = nullptr;
+        sd_bus_default(&l_sdBus);
+
+        const types::InventoryCalloutData& l_invCallout = i_callouts[0];
+        const types::CalloutPriority& l_priorityEnum = get<1>(l_invCallout);
+
+        const std::string& l_priority =
+            (m_priorityMap.find(l_priorityEnum) != m_priorityMap.end()
+                 ? m_priorityMap.at(l_priorityEnum)
+                 : m_priorityMap.at(types::CalloutPriority::Low));
+
+        auto l_rc = -1;
+        if (!i_userData2)
+        {
+            l_rc = sd_bus_call_method_async(
+                l_sdBus, NULL, constants::eventLoggingServiceName,
+                constants::eventLoggingObjectPath,
+                constants::eventLoggingInterface, "Create", NULL, NULL,
+                "ssa{ss}", l_message.c_str(), l_severity.c_str(), 6, "FileName",
+                i_fileName.c_str(), "FunctionName", i_funcName.c_str(),
+                "InternalRc", i_internalRc.c_str(), "DESCRIPTION",
```

Comment:
is should be "InternalReturnCode" or "InternalReasonCode" ? and
is "FileName" or "File Name" ?

Reply:
Did not understand why this question